examples/globus_app.py

```diff
@@ -0,0 +1,155 @@
+   `Redirect URL` as `http://localhost:5000/oauth/authorized/globus/`. See
+   here for more documentation:
+   `https://docs.globus.org/api/auth/developer-guide/`
+
+
+2. Grab the *Client ID* and *Client Secret* after registering the application
+   and add them to your instance configuration as `consumer_key` and
+   `consumer_secret`.
+
+   .. code-block:: console
+
+       $ export GLOBUS_APP_CREDENTIALS_KEY=my_globus_client_id
+       $ export GLOBUS_APP_CREDENTIALS_SECRET=my_globus_client_secret
+
+3. Create database and tables:
+
+   .. code-block:: console
+
+       $ cdvirtualenv src/invenio-oauthclient
+       $ pip install -e .[all]
+       $ cd examples
+       $ export FLASK_APP=globus_app.py
+       $ ./app-setup.py
+
+You can find the database in `examples/globus_app.db`.
+
+4. Run the development server:
+
+   .. code-block:: console
+
+       $ flask run -p 5000 -h '0.0.0.0'
+
+5. Open in a browser the page `http://localhost:5000/globus`.
+
+   You will be redirected to globus to authorize the application.
+
+   Click on `Allow` and you will be redirected back to
+   `http://localhost:5000/oauth/signup/globus/`, where you will be able to
+   finalize the local user registration.
+
+6. To clean up and drop tables:
+
+   .. code-block:: console
+
+       $ ./app-teardown.sh
+
+SPHINX-END
+
+"""
+
+from __future__ import absolute_import, print_function
+
+import os
+
+from flask import Flask, redirect, url_for
+from flask_babelex import Babel
+from flask_login import current_user
+from flask_menu import Menu as FlaskMenu
+from flask_oauthlib.client import OAuth as FlaskOAuth
+from invenio_accounts import InvenioAccounts
+from invenio_accounts.views import blueprint as blueprint_user
+from invenio_db import InvenioDB
+from invenio_mail import InvenioMail
+from invenio_userprofiles import InvenioUserProfiles
+from invenio_userprofiles.views import \
+    blueprint_ui_init as blueprint_userprofile_init
+
+from invenio_oauthclient import InvenioOAuthClient
+from invenio_oauthclient.contrib import globus
+from invenio_oauthclient.views.client import blueprint as blueprint_client
+from invenio_oauthclient.views.settings import blueprint as blueprint_settings
+
+# [ Configure application credentials ]
+GLOBUS_APP_CREDENTIALS = dict(
+    consumer_key=os.environ.get('GLOBUS_APP_CREDENTIALS_KEY'),
+    consumer_secret=os.environ.get('GLOBUS_APP_CREDENTIALS_SECRET'),
+)
+
+# Create Flask application
+app = Flask(__name__)
+
+app.config.update(
+    SQLALCHEMY_DATABASE_URI=os.environ.get(
+        'SQLALCHEMY_DATABASE_URI', 'sqlite:///globus_app.db'
+    ),
+    OAUTHCLIENT_REMOTE_APPS=dict(
+        globus=globus.REMOTE_APP,
+    ),
+    GLOBUS_APP_CREDENTIALS=GLOBUS_APP_CREDENTIALS,
+    DEBUG=True,
+    SECRET_KEY='TEST',
+    SQLALCHEMY_ECHO=False,
+    SECURITY_PASSWORD_SALT='security-password-salt',
+    MAIL_SUPPRESS_SEND=True,
+    TESTING=True,
+    USERPROFILES_EXTEND_SECURITY_FORMS=True,
```

Comment:
A couple notes on Config options. 

* Mail throws an error when a user logs in due to a bug in [Flask-Security](https://github.com/mattupstate/flask-security/issues/685). However, [Invenio is aware of the problem here](https://github.com/inveniosoftware/invenio-app-ils/issues/28). It's probably better to apply the fix in one of the dependent packages instead to avoid duplication. 
* MAIL_SUPPRESS_SEND also throws an error if it's set to `False`. This happens very rarely for logged in users and is probably better handled in downstream packages. I'm not completely sure why editing this value results in an error, but it seemed to be a [caching error here](https://github.com/mitsuhiko/flask-oauth/issues/48)
* SQLALCHEMY_TRACK_MODIFICATIONS throws a bunch of warnings, but should also probably be left to dependent packages to configure for themselves. We could set this in tests to squelch the warning messages on tests.